portlist/portlist_linux_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package portlist
+
+import (
+	"bufio"
+	"bytes"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestParsePorts(t *testing.T) {
+	tests := []struct {
+		name string
+		in   string
+		want []Port
+	}{
+		{
+			name: "empty",
+			in:   "header line (ignored)\n",
+			want: nil,
+		},
+		{
+			name: "ipv4",
+			in: `header line
+  0: 0100007F:0277 00000000:0000 0A 00000000:00000000 00:00000000 00000000     0        0 22303 1 0000000000000000 100 0 0 10 0
```

Comment:
aside: with userspace networking mode, these services shouldn't be skipped probably, as they're then accessible.

Reply:
At some point we should have an argument about whether that's correct. It's a surprising change in security boundaries to have localhost services become remotely accessible.